src/app/AuthenticationController.tsx

```diff
@@ -193,13 +197,27 @@ class AuthenticationController extends React.Component<AuthenticationControllerP
         history.replace(this.props.landingRoute);
         this.props.setLandingRoute(undefined);
       }
-      this.setState({ stage: LoginStage.LOGGED_IN });
+      // Setting postLoginError to 'undefined' here ensures that when the user logs out
+      // after logging in successfully, they do not see the previous postLoginError message.
+      this.setState({ stage: LoginStage.LOGGED_IN, postLoginError: undefined });
     } catch (err) {
       console.error('Error on post-login actions.', err);
-      this.setState({ isPostLoginError: true });
+      this.setState({ postLoginError: this.generatePostLoginErrorMessage(err) });
+      // Logging the user out immediately since they will need to log in again
+      // in order to retry the 'post-login' actions. The post-login actions
+      // must succeed before the user can proceed.
+      //
+      // When logout happens a re-render happens so need to track if the user has been previously
+      // logged out from this error message and display the message then.
+      this.props.logout();
     }
   };
 
+  private generatePostLoginErrorMessage(err: any): string {
+    return `Kiali failed to initialize due to '${API.getErrorString(err)}'. Please ensure that services 
+    Kiali depends on such as Prometheus are healthy and reachable by Kiali then login again.`;
```

Comment:
I'm happy to update this language to whatever makes the most sense. I just wanted to be sure and highlight where the problem might be such as with Prom being unreachable.

Reply:
this message is OK to me